api/envoy/config/core/v3/protocol.proto

```diff
@@ -260,6 +257,11 @@ message KeepaliveSettings {
   // A value of zero means there will be no jitter.
   // The default value is 15%.
   type.v3.Percent interval_jitter = 3;
+
+  // If the connection has been idle for this duration, send a HTTP/2 ping ahead
+  // of new stream creation, to quickly detect dead connections.
+  google.protobuf.Duration connection_idle_interval = 4
```

Comment:
Would it be worth explaining how this interacts with `interval`? Does this need to be less than `interval` to be effective?